apps/verticalhome/js/pinned_apps_navigation.js

```diff
@@ -13,274 +13,375 @@
+          }
+          else{
+          //preVerticalNavigation.call(this);
 
-            var removed = this.elements.splice(0, 1);
+          this.selectedElemIndex = this.currentIndex() || 0;
 
-            var cloned = removed[0].cloneNode(true);
 
-            removed[0].classList.add('removed');
+          if(this.selectedElemIndex<=0)
+          {
+            this.selectedElemIndex = 0;
+            this.elements[0].classList.add('selected');
+            needClone = false;
+          }
 
-            this.elemList.appendChild(cloned);
+          currSelected = this.elements[this.selectedElemIndex];
+          nextSelected = this.elements[this.selectedElemIndex+1];
 
-            if (app.pinnedAppsManager.items[+cloned.dataset.index]){
-              app.pinnedAppsManager.
-                  items[+cloned.dataset.index].
-                  refreshDomElem(cloned);
-            }
+          nextSelected.classList.add('selected');
 
-            this.elements[this.elements.length] = cloned;
+          if(needClone){
+            itm = this.elements[0];
+            cln = itm.cloneNode(true);
 
-            this.selectedElemIndex++;
-            this.elements[this.selectedElemIndex].classList.add('selected');
+            this.elemList.appendChild(cln);
+            itm.parentNode.removeChild(itm);
+          }
 
-            this.elemList.style.top = this.elemList.dataset.scrolldown;
-          } else {
-            this.selectedElemIndex++;
-            this.elements[this.selectedElemIndex].classList.add('selected');
 
-            topStyle = parseFloat(this.elemList.style.top);
-            this.elemList.style.top = topStyle - 6 + 'rem';
-            this.elemList.dataset.scrolldown = this.elemList.style.top;
+          if( currSelected.classList.contains('selected')){
+           currSelected.classList.remove('selected');
+          }
+          if( currSelected.classList.contains('middle')){
+          currSelected.classList.remove('middle');
           }
 
-          this.prevVerticalKey = 'down';
+          this.refresh();
+          this.scrollNum();
 
-          if (this.selectedElemIndex == 1) {
-            this.elemList.style.top = '0rem';
           }
-
           break;
 
         case 'Accept':
+          var ElsSelectedIndex = this.elements[this.selectedElemIndex];
 
-          window.removeEventListener('keydown', this);
+          if (this.personalizeMode){
+            if(ElsSelectedIndex.getAttribute('id') == 'addFromMoreApps'){
+                app.enterMoreAppsPersonalise();
+             }
+          }
+          else{
 
-          var elemId = this.elements[this.selectedElemIndex].getAttribute('id');
-          if ( elemId === 'moreApps') {
-            app.showMoreApps();
-          } else {
+            if (ElsSelectedIndex.getAttribute('id') == 'moreApps') {
+              app.showMoreApps();
+            } else {
+              app.getAppByURL(ElsSelectedIndex.dataset.manifesturl).launch();
```

Comment:
As I remember this function launch dialer app not correct. 
I think it would be better provides click function for this or add entry_point parameter to function launch().